flink-python/src/main/java/org/apache/flink/python/util/TarGzUtils.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.flink.python.util;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.util.IOUtils;
+
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/** Utils used to extract tar.gz files and try to restore the origin permissions of files. */
+@Internal
+public class TarGzUtils {
```

Comment:
For the convenience of review, I prefer leaving it as is in this review, and work on a simple refactoring separately. But I am ok if you think that aligns better with Flink development practice.

Reply:
Since there is already DecompressUtils file and so I'm in favor of moving the code there. Otherwise we have three files for decompression: DecompressUtils, TarGzUtils, ZipUtils. It's a little confusing for users. Since most of the code are newly added and so I think it will be all right for code review.

Reply:
will do